src/__tests__/surveys.test.ts

```diff
@@ -119,674 +129,658 @@ describe('surveys', () => {
     beforeEach(() => {
         surveysResponse = { surveys: firstSurveys }
 
-        config = {
+        const config = {
             token: 'testtoken',
             api_host: 'https://app.posthog.com',
             persistence: 'memory',
+            disable_surveys: undefined,
         } as unknown as PostHogConfig
 
-        instance = {
-            config: config,
-            persistence: new PostHogPersistence(config),
-            requestRouter: new RequestRouter({ config } as any),
-            _addCaptureHook: jest.fn(),
-            register: (props: Properties) => instance.persistence?.register(props),
-            unregister: (key: string) => instance.persistence?.unregister(key),
-            get_property: (key: string) => instance.persistence?.props[key],
-            _send_request: jest
-                .fn()
-                .mockImplementation(({ callback }) => callback({ statusCode: 200, json: surveysResponse })),
-            featureFlags: {
-                _send_request: jest
-                    .fn()
-                    .mockImplementation(({ callback }) => callback({ statusCode: 200, json: decideResponse })),
-                getFeatureFlag: jest.fn().mockImplementation((featureFlag) => decideResponse.featureFlags[featureFlag]),
-                isFeatureEnabled: jest
-                    .fn()
-                    .mockImplementation((featureFlag) => decideResponse.featureFlags[featureFlag]),
-            },
-        } as unknown as PostHog
+        instance = defaultPostHog().init('testtoken', { ...config }, uuidv7())!
 
-        surveys = new PostHogSurveys(instance)
+        loadScriptMock.mockImplementation((_path, callback) => {
+            assignableWindow.__PosthogExtensions__ = assignableWindow.__Posthog__ || {}
+            assignableWindow.extendPostHogWithSurveys = generateSurveys
+            assignableWindow.__PosthogExtensions__.SurveysEventReceiver = SurveyEventReceiver
+            assignableWindow.__PosthogExtensions__.canActivateRepeatedly = canActivateRepeatedly
+            assignableWindow.__PosthogExtensions__.hasEvents = hasEvents
 
-        Object.defineProperty(window, 'location', {
-            configurable: true,
-            enumerable: true,
-            writable: true,
-            // eslint-disable-next-line compat/compat
-            value: new URL('https://example.com'),
+            callback()
```

Comment:
this entrypoint script behavior duplication is spreading, we definitely need a nicer way of doing it